src/main/java/ti4/generator/UnitRenderGenerator.java

```diff
@@ -0,0 +1,545 @@
+
+        if (ctx.isMirage) {
+            imageDmgX = imageX - TILE_PADDING;
+            imageDmgY = imageY - TILE_PADDING;
+        } else if (unitType == UnitType.Mech) {
+            imageDmgX = getDamageX(unitPos, imagePos, dmgImage);
+            imageDmgY = getDamageY(unitPos, imagePos, dmgImage);
+        } else {
+            imageDmgX = getCenteredDamageX(unitPos, imagePos, unitImage, dmgImage);
+            imageDmgY = getCenteredDamageY(unitPos, imagePos, unitImage, dmgImage);
+        }
+
+        if (unitPos != null && ctx.isSpace) {
+            imageDmgX -= 7;
+        }
+
+        tileGraphics.drawImage(dmgImage, TILE_PADDING + imageDmgX, TILE_PADDING + imageDmgY, null);
+    }
+
+    private int getDamageX(Point position, ImagePosition imagePosition, BufferedImage dmgImage) {
+        return position != null ? position.x : imagePosition.originalX() - dmgImage.getWidth();
+    }
+
+    private int getDamageY(Point position, ImagePosition imagePosition, BufferedImage dmgImage) {
+        return position != null ? position.y : imagePosition.originalY() - dmgImage.getHeight();
+    }
+
+    private int getCenteredDamageX(Point position, ImagePosition imagePosition, BufferedImage unitImage, BufferedImage dmgImage) {
+        return position != null
+            ? position.x + (unitImage.getWidth() / 2) - (dmgImage.getWidth() / 2)
+            : imagePosition.originalX() - (dmgImage.getWidth() / 2);
+    }
+
+    private int getCenteredDamageY(Point position, ImagePosition imagePosition, BufferedImage unitImage, BufferedImage dmgImage) {
+        return position != null
+            ? position.y + (unitImage.getHeight() / 2) - (dmgImage.getHeight() / 2)
+            : imagePosition.originalY() - (dmgImage.getHeight() / 2);
+    }
+
+    private boolean shouldSkipInvalidUnit(UnitKey unitKey) {
+        return unitKey == null || !Mapper.isValidColor(unitKey.getColor());
+    }
+
+    private boolean shouldHideJailUnit(Player frogPlayer, UnitKey unitKey) {
+        if (!ctx.isJail || frogPlayer == null) {
+            return false;
+        }
+        String colorID = Mapper.getColorID(frogPlayer.getColor());
+        return !ctx.showJail && !unitKey.getColorID().equals(colorID);
+    }
+
+    private String getUnitPath(UnitKey unitKey) {
+        String unitPath = Tile.getUnitPath(unitKey);
+        if (unitPath == null) return null;
+
+        // Handle bulk unit replacements
+        unitPath = switch (unitKey.getUnitType()) {
+            case Fighter -> unitPath.replace(Constants.COLOR_FF, Constants.BULK_FF);
+            case Infantry -> unitPath.replace(Constants.COLOR_GF, Constants.BULK_GF);
+            default -> unitPath;
+        };
+
+        Player player = game.getPlayerByColorID(unitKey.getColorID()).orElse(null);
+        if (player == null) return unitPath;
+
+        // Handle special unit cases
+        if (unitKey.getUnitType() == UnitType.Spacedock && player.ownsUnitSubstring("cabal_spacedock")) {
```

Comment:
Not sure if this was finished yet, but there was a project to stop using the "csd" image for these and instead be painting the rift under the normal space dock image instead, to eliminate the need for graphics folks to create the csd version for each colour.
If that doesn't look to be handled yet, then we can leave this as is